### PR TITLE
Feature/read user

### PR DESCRIPTION
**Pull request by BoskoGogic95**
Implementation of Read operation.

### CODE REVIEW

File: BE/Api.Access.Layer/Controllers/UserController.cs [hunk 1]
```diff
@@ -32,6 +32,15 @@ public async Task<IActionResult> Delete([FromBody] string email)
 
         }
 
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        [HttpGet("read")]
+        public async Task<IActionResult> Read([FromBody] string email)
```
Comment (petar-kovac) [inline BE/Api.Access.Layer/Controllers/UserController.cs:38]: Please rename this method to 'GetUser'. Also, it is a good practice to name our methods in such a way that when you read the title you know what that method does (without the need to look at its implementation of it).

So in this case, as you get user by email parameter the name of the method should be like "GetUserByEmail".

File: BE/Business.Access.Layer/Services/UserService.cs [hunk 2]
```diff
@@ -44,9 +44,24 @@ public async Task<BusinessUserModel> Delete(string mail)
             
         }
 
-        public Task<BusinessUserModel> GetByMail(string mail)
+        public async Task<BusinessUserModel> GetByMail(string email)
         {
-            throw new NotImplementedException();
+            var userForRead = await _unitOfWork.Users.FindSingleAsync(user => user.Email.Equals(email));
```
Comment (petar-kovac) [inline BE/Business.Access.Layer/Services/UserService.cs:49]: What happens if the email parameter is null?

File: BE/Business.Access.Layer/Services/UserService.cs [hunk 3]
```diff
@@ -44,9 +44,24 @@ public async Task<BusinessUserModel> Delete(string mail)
...
+            
+            if(email == null)
+            {
+                throw new KeyNotFoundException("User email has NULL value.");
+            }
+
+            if(userForRead == null)
+            {
+                throw new KeyNotFoundException("User with that email doesn't exist!");
```
Comment (petar-kovac) [inline BE/Business.Access.Layer/Services/UserService.cs:58]: When we return exception messages we don't want to yell at our users (if the message will be presented to the end-user).
There is no need for exclamation here.

FYI (I've seen that this repeats on someplace): @danielastrbac @srdjanstanojcic @damjan21

Comment (srdjanstanojcic) [reply BE/Business.Access.Layer/Services/UserService.cs:58]: We just send this message to FE, they can edit it in best way for users. 😄

File: BE/Api.Access.Layer/Controllers/UserController.cs (hunk 1 above)
Comment (BoskoGogic95) [reply BE/Api.Access.Layer/Controllers/UserController.cs:38]: Okay, my fault, I will fix it. Thanks for suggestion about giving name of methods I will try to improve it.

File: BE/Business.Access.Layer/Services/UserService.cs (hunk 2 above)
Comment (BoskoGogic95) [reply BE/Business.Access.Layer/Services/UserService.cs:49]: I put If statement below which checks if the email is null. Maybe I just have to put it above this line ?

**Pull request merged**